src/modules/featureFlag/featureFlagUpdateChecker.ts

```diff
@@ -0,0 +1,24 @@
+import { join } from 'path'
+import { spawnUnblockingChildProcess } from '../../lib/utils/spawnUnblockingChildProcess'
+import { FeatureFlag } from './featureFlag'
+
+export class FeatureFlagUpdateChecker {
+  private static readonly FEATURE_FLAG_CHECK_INTERVAL = 1000 * 60 * 2 // 2 Minutes
+
+  public static checkForUpdateFeatureFlag() {
+    if (FeatureFlagUpdateChecker.shouldUpdateFeatureFlagFile()) {
+      FeatureFlagUpdateChecker.startUpdateFileProcess()
+    }
+  }
+
+  private static shouldUpdateFeatureFlagFile() {
+    return (
+      Date.now() - FeatureFlag.getSingleton().getLastFeatureFlagUpdate() >=
+      FeatureFlagUpdateChecker.FEATURE_FLAG_CHECK_INTERVAL
+    )
```

Comment:
_suggestion_
You can make it a little more readable:
```
const elapsedTime = Date.now() - FeatureFlag.getSingleton().getLastFeatureFlagUpdate()
return elapsedTime >= FeatureFlagUpdateChecker.FEATURE_FLAG_CHECK_INTERVAR
```

`elapsedTime` is not a great name, maybe you can think of something better